website/docs/deployment.mdx

```diff
@@ -54,6 +54,50 @@ Use [slorber/trailing-slash-guide](https://github.com/slorber/trailing-slash-gui
 
 :::
 
+## Choosing a hosting provider
+
+There are a few common hosting options:
+
+- [Self hosting](#self-hosting) with an HTTP server like Apache2 or Nginx;
+- Jamstack providers, e.g. [Netlify](#deploying-to-netlify) and [Vercel](#deploying-to-vercel). We will use Netlify as reference, but the same reasoning can apply to other providers.
+- [GitHub Pages](#deploying-to-github-pages). (By definition, it is also Jamstack, but we compare it separately.)
+
+If you are unsure of which one to choose, ask the following questions:
+
+<details>
+
+<summary>
+  How much resource (person-hours, money) am I willing to invest in this?
+</summary>
+
+- 🔴 Self-hosting is the hardest to set up—you would usually need an experienced person to manage this. Cloud services is almost never free, and setting up an on-site server and connecting it to the WAN can be more costly.
+- 🟢 Jamstack providers can help you set up a working website in almost no time and offers features like server-side redirects that are easily configurable. Build time can be limited—you would need to pay in order to get more build time quota. Check the pricing page of your provider for details.
+- 🟡 The GitHub Pages deployment workflow can be tedious to set up. (Evidence: compare the length of [Deploying to Netlify](#deploying-to-netlify) and [Deploying to GitHub Pages](#deploying-to-github-pages)!) However, this service (including build and deployment) is always free, and we have detailed instructions to help you make it work.
```

Comment:
maybe just mention the length got the GH instructions, not Netlify?

Also it's not 100% free either, it's free if you run deploy locally manually, not if you automate with GH actions

I would mostly mention that GH pages is a jamstack provider that gives you very limited server options

Reply:
GitHub actions is entirely free. Public repos have unlimited build time: https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

> GitHub Actions usage is free for both public repositories and self-hosted runners. For private repositories, each GitHub account receives a certain amount of free minutes and storage,